src/Dictionaries/CacheDictionaryUpdateQueue.cpp

```diff
@@ -133,19 +133,23 @@ void CacheDictionaryUpdateQueue<dictionary_key_type>::updateThreadFunction()
             /// Update
             update_func(unit_to_update);
 
-            /// Notify thread about finished updating the bunch of ids
-            /// where their own ids were included.
-            std::lock_guard lock(update_mutex);
+            {
+                /// Notify thread about finished updating the bunch of ids
+                /// where their own ids were included.
+                std::lock_guard lock(unit_to_update->update_mutex);
+                unit_to_update->is_done = true;
```

Comment:
`is_done` is atomic, why do we need the lock here?

Reply:
Even if conditional variable state changes are atomic, state change and state check before wait must be synchronized, otherwise there could be potential lost wakeup.